### PR TITLE
Conditionally bind mount /usr/bin/docker-current when it is present

### PR DESCRIPTION
**Pull request by detiber**
Resolves: https://github.com/openshift/openshift-ansible/issues/1921

### CODE REVIEW

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
@sdodson @abutcher ptal

I'm seeing the following error in my node logs, but I'm not sure if it's related to this PR or a different issue yet:
```
May 20 16:36:41 ose3-master1-ansible.test.example.com atomic-openshift-node[126655]: W0520 16:36:41.291139  126792 container_manager_linux.go:278] [ContainerManager] Failed to ensure state of "/docker-daemon": failed to find pid of Docker container: exit status 1
```

**Comment by detiber** [comment]
On closer inspection, it looks like my deployment installed version 3.1.1.6 instead of 3.2, so I suspect the error I'm seeing is not related.

**Comment by detiber** [comment]
aos-ci-test-extended

**Comment by detiber** [comment]
@abutcher @sdodson ptal

**Comment by sdodson** [comment]
LGTM

**Comment by abutcher** [comment]
lgtm

**Comment by abutcher** [comment]
We should add `/etc/sysconfig/{{ openshift.common.service_type }}-node-dep` to the uninstall playbook.

**Comment by sdodson** [comment]
`/etc/systemd/system/{{ openshift.common.service_type }}-node-dep.service` too

**Comment by detiber** [comment]
aos-ci-test

**Comment by arilivigni** [comment]
aos-ci-test

**Comment by arilivigni** [comment]
aos-ci-test

**Comment by arilivigni** [comment]
aos-ci-test

**Comment by arilivigni** [comment]
aos-ci-test

**Comment by sdodson** [comment]
@detiber Docker is failing to start when the variable is empty

File: roles/openshift_node/templates/openshift.docker.node.dep.service [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Requires=docker.service
+After=docker.service
+PartOf={{ openshift.common.service_type }}-node.service
+Before={{ openshift.common.service_type }}-node.service
+
+
+[Service]
+ExecStart=/bin/bash -c "if [[ -f /usr/bin/docker-current ]]; then echo \"DOCKER_ADDTL_BIND_MOUNTS=--volume=/usr/bin/docker-current:/usr/bin/docker-current:ro\" > /etc/sysconfig/{{ openshift.common.service_type }}-node-dep; else echo \"DOCKER_ADDTL_BIND_MOUNTS=\" > /etc/sysconfig/{{ openshift.common.service_type }}-node-dep; fi"
```
Comment (sdodson) [inline roles/openshift_node/templates/openshift.docker.node.dep.service]: Switch to commenting it on the negative

File: roles/openshift_node/templates/openshift.docker.node.service [hunk 2]
```diff
@@ -8,11 +8,14 @@ Requires=docker.service
...
+ExecStart=/usr/bin/docker run --name {{ openshift.common.service_type }}-node --rm --privileged --net=host --pid=host --env-file=/etc/sysconfig/{{ openshift.common.service_type }}-node -v /:/rootfs:ro -e CONFIG_FILE=${CONFIG_FILE} -e OPTIONS=${OPTIONS} -e HOST=/rootfs -e HOST_ETC=/host-etc -v {{ openshift.common.data_dir }}:{{ openshift.common.data_dir }} -v {{ openshift.common.config_base }}/node:{{ openshift.common.config_base }}/node {% if 'cloudprovider' in openshift and 'kind' in openshift.cloudprovider and openshift.cloudprovider.kind != '' -%} -v {{ openshift.common.config_base }}/cloudprovider:{{ openshift.common.config_base}}/cloudprovider {% endif -%} -v /etc/localtime:/etc/localtime:ro -v /etc/machine-id:/etc/machine-id:ro -v /run:/run -v /sys:/sys:ro -v /usr/bin/docker:/usr/bin/docker:ro -v /var/lib/docker:/var/lib/docker -v /lib/modules:/lib/modules -v /etc/origin/openvswitch:/etc/openvswitch -v /etc/origin/sdn:/etc/openshift-sdn -v /etc/systemd/system:/host-etc/systemd/system -v /var/log:/var/log -v /dev:/dev ${DOCKER_ADDTL_BIND_MOUNTS} {{ openshift.node.node_image }}:${IMAGE_VERSION}
```
Comment (sdodson) [inline roles/openshift_node/templates/openshift.docker.node.service]: and removing remove braces from  ${DOCKER_ADDTL_BIND_MOUNTS} fixed both positive and negative conditions for me

**Comment by sdodson** [comment]
My test was to install containerized, then downgrade to docker-1.9.1-25 `yum swap docker-* docker-*1.9.1-25*` and restart the node.

**Comment by sdodson** [comment]
Superseded by #1941

**Pull request closed without merging**